### PR TITLE
Remove render_test_results job

### PR DESCRIPTION
**Pull request by driazati**
This removes the `render_test_results` job we had before which had been causing some confusion among devs when it failed and isn't really necessary now that we can actually render test results on the PR HUD.

### CODE REVIEW

File: .github/templates/bazel_ci_workflow.yml.j2 [hunk 1]
```diff
@@ -156,71 +157,5 @@ on:
           if-no-files-found: error
           path:
             test-reports-*.zip
-      - name: Clean up docker images
-        if: always()
-        run: |
-          # Prune all of the docker images
-          docker system prune -af
-{%- endblock %}
```
Comment (janeyx99) [inline .github/templates/bazel_ci_workflow.yml.j2:164]: Is this removal intentional?

Comment (driazati) [reply .github/templates/bazel_ci_workflow.yml.j2]: nope, fixed

**Review by seemethere — APPROVED** [review]
LGTM, use all of the macros!

**Review by walterddr — APPROVED** [review]
this looks great. btw would it also work with this type of failures --> https://github.com/pytorch/pytorch/actions/runs/1164148764

**Review by janeyx99 — APPROVED** [review]

**Review by seemethere — APPROVED** [review]
